### ISSUE TITLE
`rpg_character_sheet` analysis doesn't detect use of `IO.examine` if you pipe into it

### ISSUE THREAD

**Issue by hunkyjimpjorps** [exercism/elixir-analyzer#144]
For the final task, it looks like the analyzer is expecting an independent call to `IO.examine` and doesn't catch piping into it.

https://github.com/exercism/elixir-analyzer/blob/290d206c62ae04d5cc9e5bd05763133761899db5/lib/elixir_analyzer/test_suite/rpg_character_sheet.ex#L89-L95

My submitted code:
```elixir
  def run() do
    welcome()
    character_map = %{name: ask_name()} |>
      Map.put(:class, ask_class()) |>
      Map.put(:level, ask_level()) |>
      IO.inspect(label: "Your character:")
  end
```

Exercism's feedback: 

![image](https://user-images.githubusercontent.com/5170341/132401295-c3a7656b-943b-4064-b1f2-50c0cd9af2f4.png)

**Comment by angelikatyborska** [comment]
Thank you!

@jiegillet This sounds like something that _should have_ been fixed by https://github.com/exercism/elixir-analyzer/pull/140

Can you take a look?

**Comment by neenjaw** [comment]
The pipe is wrapped with a match (`=`), so I don't think #140 would catch that

**Comment by jiegillet** [comment]
Exactly this ⬆️.

I haven't had time to double check, but I think an assignment on the last line should always get you a compiler warning. I can add a common check that checks for that situation, I think that would be a nice addition. This would probably depend on the machinery I set up in #133 though.

**Comment by neenjaw** [comment]
Yes, there is a compiler warning for that since it is an "unused var"

**Comment by angelikatyborska** [comment]
But we don't have a way yet to communicate compiler warnings 🙁 sorry, I didn't have time to review #133 yet and it might not happen for another few days.
